src/diffTool/diffProcessor.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/* eslint-disable @typescript-eslint/no-use-before-define */
```

Comment:
I have placed all the functions so that its easy to read the code and flow. But typescript wants all the functions defined before it is used and hence the disabled rule. https://palantir.github.io/tslint/rules/no-use-before-declare/